pkg/xds/server/callbacks/dataplane_callbacks.go

```diff
@@ -104,14 +146,27 @@ func (d *xdsCallbacks) OnStreamRequest(streamID core_xds.StreamID, request util_
 	// in case client will open 2 concurrent request for the same streamID then
 	// we don't to increment the counter twice, so checking once again that stream
 	// wasn't processed
-	alreadyProcessed = d.dpStreams[streamID].dp != nil
+	if isDelta {
+		alreadyProcessed = d.dpDeltaStreams[streamID].dp != nil
+	} else {
+		alreadyProcessed = d.dpStreams[streamID].dp != nil
+	}
 	if alreadyProcessed {
 		return nil
 	}
 
-	dpStream := d.dpStreams[streamID]
+	var dpStream dpStream
+	if isDelta {
+		dpStream = d.dpDeltaStreams[streamID]
+	} else {
+		dpStream = d.dpStreams[streamID]
+	}
 	dpStream.dp = &dpKey
-	d.dpStreams[streamID] = dpStream
+	if isDelta {
+		d.dpDeltaStreams[streamID] = dpStream
+	} else {
+		d.dpStreams[streamID] = dpStream
+	}
```

Comment:
maybe instead of this constant if/else you could have a generic function that takes either `dpStreams` or `dpDeltaStreams`? this applies to all functions with `isDelta`